pkg/operator/ceph/spec/spec.go

```diff
@@ -139,3 +140,29 @@ func PodLabels(appName, namespace, daemonType, daemonID string) map[string]strin
 	labels[daemonType] = daemonID
 	return labels
 }
+
+// CheckPodMemory verify pod's memory limit is valid
+func CheckPodMemory(resources v1.ResourceRequirements, cephPodMinimumMemory uint64) error {
+	// Ceph related PR: https://github.com/ceph/ceph/pull/26856
+	podMemoryLimit := resources.Limits.Memory()
+	podMemoryRequest := resources.Requests.Memory()
+	errorMessage := `refuse to run the pod with %dmb of ram, provide at least %dmb`
+
+	// If nothing was provided let's just return
+	// This means no restrictions on pod's resources
+	if podMemoryLimit.IsZero() && podMemoryRequest.IsZero() {
+		return nil
+	}
+
+	// This means LIMIT and REQUEST are either identical or different but still we use LIMIT as a reference
+	if uint64(podMemoryLimit.Value()) < display.MbTob(cephPodMinimumMemory) {
+		return fmt.Errorf(errorMessage, display.BToMb(uint64(podMemoryLimit.Value())), cephPodMinimumMemory)
+	}
+
+	// FYI: there is no such case as LIMIT inferior to REQUEST
```

Comment:
Although we don't *have* to handle this case, I wonder if it would be a better user experience to handle it and inform the user of the issue explicitly. @travisn thoughts?

Reply:
If the pod fails to start, it will likely be more noticeable than an error message in the operator log, right? Then they can troubleshoot with kubectl instead of opening the log

Reply:
The thing is we will never reach that code if LIMIT is inferior to REQUEST, k8s will refuse to schedule that pod.

Reply:
I have 2 issues here. The first is that the comment is inherently wrong. The case of limit being lower than request *does* exist, because the user can specify those inputs in that way. The case that does not exist is Kubernetes successfully scheduling a pod with those inputs.

Second, I don't think that we should always avoid error messages when Kubernetes will report a failure. Especially if Rook's error message is helpful, we will be adding value for the users, and in my experience, Kubernetes' error messages are not always as helpful as one would hope. 

Consider: `logger.Errorf("user has specified a pod memory limit below the pod memory request in the cluster CRD. Rook will create pods that are expected to fail to serve as a more apparent error indicator to the user.")` This message informs the user (1) exactly what the issue is in plain terms, (2) that it is a user configuration issue, and (3) where to go to fix the configuration issue. The kubernetes issue the user is likely to get will be that resource requests are invalid, but the users doesn't know if they have configured something improperly or if Rook has a bug.